trainer.py

```diff
@@ -122,17 +123,24 @@ def _prune(self, model):
     def prune_trained_model(model, threshold):
         if math.isnan(threshold):
             return
+        v = Verbose()
         # Layer 0 has no input edges, start from layer 1
         pruned_edges = 0
         for i in range(1, len(model.layers)):
             weights = model.layers[i].get_weights()
             input_weights = weights[0]  # weights[1] is the list of node biases
+            weighted_threshold = threshold * np.linalg.norm(input_weights)
             # The incoming edge weights of node N is incoming_edge_weights[N]
-            new_weights = np.where((input_weights < threshold) & (input_weights > -threshold), 0, input_weights)
-            pruned_edges += new_weights.size - np.count_nonzero(new_weights)
+            new_weights = np.where(np.absolute(input_weights) < weighted_threshold, 0, input_weights)
```

Comment:
you can probably use a better name for this :)

Reply:
hhhh yeah I guess it could be `pruned_weights`

Reply:
I'll do it right after you change the names of the models from `a,b`